src/views/Profile/PersonalInfo/PersonalInfo.tsx

```diff
@@ -188,11 +204,33 @@ export const PersonalInfo: React.FC<IEditAuthorProps> = ({
     setVisitFields({ ...visitFields, [inputIdentifier]: true });
   };
 
-  const buttonClickHandler = () => {
-    // TODO dispatch new author to server
+  const buttonClickHandler = async (
+    e: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    e.preventDefault();
+    try {
+      setIsLoading(true);
+      const response = author
+        ? await updateAuthorById({ authorId: author.id, ...newAuthorValues })
+        : await createAuthor(newAuthorValues);
+      if (isCurrentUser) {
```

Comment:
I did not notice this during first review round, but now I'm wondering may we want to pass `boundGetUserAsyncAction()` or `history.push(`experts/${response.data.id}`);` as a prop(e.g. `props.onSaveSuccessful`) so in the component we would need just to check `author ? update() : create()` but not to sniff if it's also a current user or not

More flexible component, more straightforward code === better readability

Do you agree?